aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -122,7 +122,7 @@ Tensor& set_storage_quantized_(
     IntArrayRef sizes,
     IntArrayRef strides) {
   auto* self_ = self.unsafeGetTensorImpl();
-  self_->set_storage(storage);
+  self_->set_storage_keep_dtype(storage);
```

Comment:
Need to verify that caller of this function didn't expect `storage` to set the dtype

Reply:
Currently the only caller of `set_storage_quantized_()` is the dispatch generated for `set_()` from `native_functions.yaml`. That dispatch goes through `THPVariable_Set()` in `tools/autograd/templates/python_variable_methods.cpp`, which has a `TORCH_INTERNAL_ASSERT` (which I'm going to change back to a `TORCH_CHECK`) to make sure the dtype of the THPStorage matches the Tensor's dtype.